packages/rocketchat-file-upload/server/lib/FileUpload.js

```diff
@@ -77,7 +78,20 @@ Object.assign(FileUpload, {
 		}));
 		return future.wait();
 	},
+	resizeImagePreview(file, callback) {
 
+		const image = FileUpload.getStore('Uploads')._store.getReadStream(file._id, file);
+
+		const transformer = sharp().resize(50, 50).max().toBuffer(function(err, out) {
+			if (err) { throw err; }
+			callback(out.toString('base64'));
+		});
+		if (/^image\/.+/.test(file.type)) {
+			image.pipe(transformer);
+		} else {
+			callback();
+		}
+	},
```

Comment:
You can create a new method like `resizeImagePreviewSync` that does not receive the callback and do the wrapAsync inside

Reply:
ok, i will do that!

Reply:
I tried to do that, but the file does not finish to send, might i'm doing something wrong.
I think the Bradley suggestion that uses Promises is nice, but i don't know if you like that..
What do you think?